src/main/java/com/atlauncher/gui/tabs/accounts/AccountsTab.java

```diff
@@ -0,0 +1,553 @@
+                            "Make sure you only do this on a " +
+                                "computer you trust.<br/>If you do this " +
+                                "on a shared computer, your password may be " +
+                                "stolen.<br/>Do you still want to save " +
+                                "your password?"))
+                        .build())
+                    .setType(DialogManager.ERROR)
+                    .addOption(GetText.tr("Yes"), true)
+                    .addOption(GetText.tr("No"))
+                    .show();
+
+                if (ret != 0) {
+                    rememberField.setSelected(false);
+                }
+            }
+        });
+
+        gbc.gridx = 0;
+        gbc.gridy++;
+        gbc.gridwidth = 2;
+        Insets BOTTOM_INSETS = new Insets(10, 0, 0, 0);
+        gbc.insets = BOTTOM_INSETS;
+        gbc.anchor = GridBagConstraints.CENTER;
+        JPanel buttons = new JPanel();
+        buttons.setLayout(new FlowLayout());
+        leftButton = new JButton(GetText.tr("Add"));
+        leftButton.setName("leftButton");
+        leftButton.addActionListener(e -> login());
+        rightButton = new JButton(GetText.tr("Clear"));
+        rightButton.addActionListener(e -> {
+            if (accountsComboBox.getSelectedIndex() == 0) {
+                clearLogin();
+            } else {
+                int ret = DialogManager
+                    .yesNoDialog()
+                    .setTitle(GetText.tr("Delete"))
+                    .setContent(GetText.tr("Are you sure you want " +
+                        "to delete this account?"))
+                    .setType(DialogManager.WARNING).show();
+                if (ret == DialogManager.YES_OPTION) {
+                    viewModel.deleteAccount();
+                }
+            }
+        });
+        loginWithMicrosoftButton =
+            new JButton(GetText.tr("Login with Microsoft"));
+        loginWithMicrosoftButton.addActionListener(e -> {
+            // TODO This should be handled by some reaction via listener
+            int numberOfAccountsBefore = viewModel.accountCount();
+            new LoginWithMicrosoftDialog();
+
+            if (numberOfAccountsBefore !=  viewModel.accountCount()) {
+                viewModel.pushNewAccounts();
+                accountsComboBox.setSelectedItem(AccountManager.getSelectedAccount());
+            }
+        });
+        buttons.add(leftButton);
+        buttons.add(rightButton);
+        buttons.add(loginWithMicrosoftButton);
+        bottomPanel.add(buttons, gbc);
+
+        rightPanel.add(topPanel, BorderLayout.NORTH);
+        rightPanel.add(bottomPanel, BorderLayout.CENTER);
+
+        contextMenu = new JPopupMenu();
+
+        updateSkin = new JMenuItem(GetText.tr("Reload Skin"));
+        updateSkin.addActionListener(e -> {
+            viewModel.updateSkin();
+
+            // Have this done via listener
```

Comment:
If this is a TODO best to mark it as one

Reply:
Fixed with [d781719](https://github.com/ATLauncher/ATLauncher/pull/607/commits/d78171975999425a923b91e5c8f21a777bfff852) and also further detailed what I mean.